framework/transport/ieee1905_transport/ieee1905_transport.cpp

```diff
@@ -6,94 +6,91 @@
+    LOG_IF(!m_broker, FATAL) << "Failed creating broker server!";
 
-    // init Local Bus interface, subscribe to specific topics
-    local_bus_ = new LocalBusInterface(Context::Instance());
-    local_bus_->Init();
-    if (local_bus_->subscriber().Subscribe<CmduTxMessage>() < 0) {
-        MAPF_ERR("cannot subscribe to local bus.");
-        return;
-    }
-    if (local_bus_->subscriber().Subscribe<InterfaceConfigurationRequestMessage>() < 0) {
-        MAPF_ERR("cannot subscribe to local bus.");
-        return;
-    }
-    local_bus_->Sync();
-    poller_.Add(local_bus_->subscriber());
+    LOG(INFO) << "starting 1905 transport.";
+
+    // Register broker handlers for internal and external messages
+    m_broker->register_external_message_handler(
+        [&](std::unique_ptr<mapf::Message> &msg, BrokerServer &broker) -> bool {
+            LOG(DEBUG) << "Processing external message: " << msg->type();
+            handle_broker_pollin_event(msg);
+            return true;
+        });
+
+    m_broker->register_internal_message_handler(
+        [&](std::unique_ptr<mapf::Message> &msg, BrokerServer &broker) -> bool {
+            LOG(DEBUG) << "Processing internal message: " << msg->type();
+            handle_broker_pollin_event(msg);
+            return true;
+        });
 
     // init netlink socket
     if (!open_netlink_socket()) {
         MAPF_ERR("cannot open netlink socket.");
         return;
     }
-    poller_.Add(netlink_fd_);
-
-    // ---------------------------------
-    // Poller Items         Type
-    // ---------------------------------
-    // local bus            mapf::Socket
-    // netlink socket       fd
-    // interface socket(s)  fd
-    // ---------------------------------
-    while (1) {
-        MAPF_DBG("polling...");
-        int nready = poller_.Poll();
-        MAPF_DBG("poll returned " << nready << ".");
-
-        // check for events on the Local Bus socket
-        MAPF_DBG("check for events on local bus subscriber socket.");
-        auto revents = poller_.CheckEvent(local_bus_->subscriber());
-        if (revents & MAPF_POLLIN) {
-            handle_local_bus_pollin_event();
-        }
-        if (revents & MAPF_POLLERR) {
-            MAPF_ERR("poll error on local bus subscriber socket.");
-            // TODO: handle this
-        }
 
-        // check for events on the netlink socket
-        MAPF_DBG("check for events on netlink socket.");
-        revents = poller_.CheckEvent(netlink_fd_);
-        if (revents & MAPF_POLLIN) {
-            MAPF_DBG("got MAPF_POLLIN event on netlink socket.");
-            handle_netlink_pollin_event();
-        }
-        if (revents & MAPF_POLLERR) {
-            MAPF_ERR("got MAPF_POLLERR event on netlink socket.");
-            // TODO: handle this
+    // Create a shared_ptr socket wrapper for the netlink socket
+    auto netlink_socket = std::shared_ptr<Socket>(new Socket(netlink_fd_), [](Socket *socket) {
```

Comment:
(nitpick)
can replace `netlink_fd_` to `m_netlink_fd`

Reply:
There's a bunch of other member variables that should be renamed as well.
This PR is big enough even without these changes :)